observability/src/main/java/io/grpc/observability/logging/CloudLoggingHandler.java

```diff
@@ -0,0 +1,190 @@
+   * @param logName the name of the log to which log entries are written
+   * @param overrideProjectId the value of cloud project id to which logs are sent to by the custom
+   *     logger
+   */
+  public CloudLoggingHandler(Level level, String logName, String overrideProjectId) {
+    baseLevel = level.equals(DEFAULT_LOG_LEVEL) ? Level.FINEST : level;
+    setLevel(baseLevel);
+
+    cloudLogName = logName != null ? logName : DEFAULT_LOG_NAME;
+
+    try {
+      // TODO(dnvindhya) read the value from config instead of taking it as an argument
+      String projectId = overrideProjectId != null ? overrideProjectId : null;
+
+      if (!Strings.isNullOrEmpty(projectId)) {
+        loggingOptions = LoggingOptions.newBuilder().setProjectId(overrideProjectId).build();
+      } else {
+        loggingOptions = LoggingOptions.getDefaultInstance();
+      }
+
+      loggingClient = loggingOptions.getService();
+    } catch (Exception ex) {
+      reportError("Error in initializing Cloud Logging", ex, ErrorManager.OPEN_FAILURE);
+      throw ex;
+    }
+  }
+
+  @Override
+  public void publish(LogRecord record) {
+    if (!(record instanceof LogRecordExtension)) {
+      throw new IllegalArgumentException("Expected record of type LogRecordExtension");
+    }
+
+    Level logLevel = record.getLevel();
+    GrpcLogRecord protoRecord = ((LogRecordExtension) record).getGrpcLogRecord();
+
+    writeLog(protoRecord, logLevel);
+  }
+
+  private void writeLog(GrpcLogRecord logProto, Level logLevel) {
+    try {
+      Severity cloudLogLevel = getCloudLoggingLevel(logLevel);
+      Map<String, Object> mapPayload = protoToMapConverter(logProto);
+
+      LogEntry grpcLogEntry =
+          LogEntry.newBuilder(JsonPayload.of(mapPayload))
+              .setSeverity(cloudLogLevel)
+              .setLogName(cloudLogName)
+              .setResource(MonitoredResource.newBuilder("global").build())
+              .build();
+
+      loggingClient.write(Collections.singleton(grpcLogEntry));
+    } catch (Exception ex) {
+      getErrorManager().error("Error in writing logs to cloud logging", ex,
+          ErrorManager.WRITE_FAILURE);
+    }
+  }
+
+  /* Google Cloud Logging API only accepts String payload or Json Payload
+  Adding all the grpc o11y logging values as string makes it difficult to look up or index
+  Hence we are going with the Json Payload approach
+   */
+  @SuppressWarnings("unchecked")
+  private Map<String, Object> protoToMapConverter(GrpcLogRecord logProto)
+      throws InvalidProtocolBufferException, IOException {
+    JsonFormat.Printer printer = JsonFormat.printer().preservingProtoFieldNames();
+    String recordJson = printer.print(logProto);
+
+    return (Map<String, Object>) JsonParser.parse(recordJson);
+  }
+
+  @Override
+  public void flush() {
+    loggingClient.flush();
+  }
+
+  @Override
+  public synchronized void close() throws SecurityException {
+    if (loggingClient != null) {
```

Comment:
Can we be consistent in null checks? Why here but not for flush()? Also if `loggingClient` is null this is a no-op instead of letting the caller know (via an exception?)